hedera-mirror-rest/topicmessage.js

```diff
@@ -189,6 +279,35 @@ const extractSqlFromTopicMessagesRequest = (topicId, filters) => {
     pgSqlParams.push(filter.value);
   }
 
+  // Query the topic_message_lookup table only for V2
+  if (config.query.v2.topicMessageLookups) {
+    // If there is no sequence number in the request URL,
+    // query the topic_message_lookup table for either the min or max sequence number depending on order.
+    if (!hasSequenceNumberForV2) {
+      if (order === constants.orderFilterValues.ASC) {
+        pgSqlQuery += ` and sequence_number >= (select MIN(lower(sequence_number_range)) 
+      from topic_message_lookup where ${TopicMessageLookup.TOPIC_ID} = $1)`;
+      } else {
+        pgSqlQuery += ` and sequence_number <= (select MAX(upper(sequence_number_range)) 
+      from topic_message_lookup where ${TopicMessageLookup.TOPIC_ID} = $1)`;
+      }
+    } else {
+      // this needs to be fixed and gotten values
+      const rows = await getTopicMessageTimestamps(topicId, filters);
+      if (rows === null) {
+        return {};
+      }
+      pgSqlQuery += ` and ((${TopicMessage.CONSENSUS_TIMESTAMP} >= ${_.first(rows).timestamp_start} and ${
+        TopicMessage.CONSENSUS_TIMESTAMP
+      } < ${_.first(rows).timestamp_end})`;
+      rows.shift();
+      rows.map((row) => {
+        pgSqlQuery += ` or (${TopicMessage.CONSENSUS_TIMESTAMP} >= ${row.timestamp_start} and ${TopicMessage.CONSENSUS_TIMESTAMP} < ${row.timestamp_end})`;
+      });
+      pgSqlQuery += `)`;
```

Comment:
construct the timestamp range conditions using functional programming patteren
```suggestion
  if (rows.length) {
     const condition = rows.map((row) => `(consensus_timestamp >= ${...} and consensus_timestamp < ${...})`).join(' or ');
     pgSqlQuery += ` and (${condition})`;
   }
```

Reply:
I just merged this with git. will fix it.